src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -440,13 +440,15 @@ protected final void retrieve(@CheckForNull SCMSourceCriteria criteria,
             String fullName = repoOwner + "/" + repository;
             final GHRepository repo = github.getRepository(fullName);
             listener.getLogger().format("Looking up %s%n", HyperlinkNote.encodeTo(repo.getHtmlUrl().toString(), fullName));
-            try {
-                repositoryUrl = repo.getHtmlUrl();
-                collaboratorNames = new HashSet<>(repo.getCollaboratorNames());
-            } catch (FileNotFoundException e) {
-                // not permitted
-                listener.getLogger().println("Not permitted to query list of collaborators, assuming none");
-                collaboratorNames = Collections.emptySet();
+            if (!repo.isPrivate()) { // random people cannot create pull requests to private repositories
```

Comment:
Just because a repo is private does not mean that all people who can create PRs are allowed to write to it, which means that a PR could be used to affect say deployments to production if we merge this PR